quinn-proto/src/connection/mod.rs

```diff
@@ -70,8 +70,18 @@ where
     handshake_cid: ConnectionId,
     /// The CID the peer initially chose, for use during the handshake
     rem_handshake_cid: ConnectionId,
+    /// Cid rotation
+    ///
+    /// Expiration timestamp of local connection IDs
+    cids_timeline: VecDeque<CidTimeStamp>,
+    /// The lifetime of Local connection IDs
+    cids_timeout: Option<Duration>,
```

Comment:
> If, as I suspect, we don't need that level of flexibility, ....

I agree. 1. I don't think current RFC ever discusses to use a different duration to retire each CID or not which gives some flexibility in implementation. 2. A different duration will cause extra efforts to sort when each cid should be retired. 3. So far, the application scenario doesn't require a different duration for each CID.

So.... as you suggested, i can consider how to move it to either `TransportConfig`, or `EndpointConfig`

Reply:
> we should remove the Option<Duration> from ConnectionIdGenerator entirely

However, i do want to argue against this point. I think `ConnectionIdGenerator` should own all fields related to CID so that a different lifetime can be applied depending on the implementation of `ConnectionIdGenerator`  trait object. Maybe `TransportConfig`, or `EndpointConfig` can be set by calling an API from `ConnectionIdGenerator`?